### PR TITLE
refactor: Improving `DateBone.singleValueFromClient()`

### PR DESCRIPTION
**Pull request by phorward**
Avoid `rawValue` and `str()`-conversion all the time.
This is a PR draft for now, as I want to test it first.

### CODE REVIEW

File: core/bones/date.py [hunk 1]
```diff
@@ -248,7 +254,7 @@ def guessTimeZone(self):
             return pytz.utc
 
         if conf["viur.instance.is_dev_server"]:
-            return tzlocal.get_localzone()
+            return pytz.timezone(tzlocal.get_localzone().key)
```
Comment (sveneberth) [inline core/bones/date.py:257]: This doesn't work for me.

```py
[2023-07-14 17:25:28,483] viur/core/request.py:300 [ERROR] '_PytzShimTimezone' object has no attribute 'key'
Traceback (most recent call last):
  File "/...deploy/viur/core/request.py", line 275, in processRequest
    self.findAndCall(path)
  File "/...deploy/viur/core/request.py", line 631, in findAndCall
    res = caller(*newArgs, **newKwargs)
  File "/...deploy/viur/core/prototypes/list.py", line 213, in edit
    return self.render.edit(skel)
  File "/...deploy/viur/core/render/json/default.py", line 212, in edit
    return self.renderEntry(skel, action, params)
  File "/...deploy/viur/core/render/json/default.py", line 152, in renderEntry
    vals = self.renderSkelValues(skel)
  File "/...deploy/viur/core/render/json/default.py", line 136, in renderSkelValues
    res[key] = self.renderBoneValue(bone, skel, key)
  File "/...deploy/viur/core/render/json/default.py", line 103, in renderBoneValue
    boneVal = skel[key]
  File "/...deploy/viur/core/skeleton.py", line 198, in __getitem__
    boneInstance.unserialize(self, key)
  File "/...deploy/viur/core/bones/base.py", line 830, in unserialize
    res = self.singleValueUnserialize(loadVal)
  File "/...deploy/viur/core/bones/date.py", line 327, in singleValueUnserialize
    time_zone = self.guessTimeZone()
  File "/...deploy/viur/core/bones/date.py", line 253, in guessTimeZone
    print(f"{tzlocal.get_localzone().key = }")
AttributeError: '_PytzShimTimezone' object has no attribute 'key'
```

While the old statement seems to return a correct pytz object:
```py
tzlocal.get_localzone() = _PytzShimTimezone(zoneinfo.ZoneInfo(key='Europe/Berlin'), 'Europe/Berlin')
```

Comment (phorward) [reply core/bones/date.py:257]: > While the old statement seems to return a correct pytz object:
> 
> ```python
> tzlocal.get_localzone() = _PytzShimTimezone(zoneinfo.ZoneInfo(key='Europe/Berlin'), 'Europe/Berlin')
> ```

Something is wrong. Did you test in on a clean environment? 

The previous code raised this, which was the reason why it was changed. 
```
[2023-06-29 15:58:54,407] /.../viur/core/request.py:299 [ERROR] ViUR has caught an unhandled exception!
[2023-06-29 15:58:54,408] /.../viur/core/request.py:300 [ERROR] 'zoneinfo.ZoneInfo' object has no attribute 'localize'
Traceback (most recent call last):
  File "/Users/achim/.local/share/virtualenvs/project-viur3-kGucZxTM/lib/python3.11/site-packages/viur/core/request.py", line 275, in processRequest
    self.findAndCall(path)
  File "/Users/achim/.local/share/virtualenvs/project-viur3-kGucZxTM/lib/python3.11/site-packages/viur/core/request.py", line 631, in findAndCall
    res = caller(*newArgs, **newKwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/achim/Arbeit/viurProjects/project-viur3/deploy/modules/user.py", line 585, in edit
    or not skel.fromClient(kwargs)  # failure on reading into the bones
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/achim/.local/share/virtualenvs/project-viur3-kGucZxTM/lib/python3.11/site-packages/viur/core/skeleton.py", line 706, in fromClient
    complete = super().fromClient(skelValues, data)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/achim/.local/share/virtualenvs/project-viur3-kGucZxTM/lib/python3.11/site-packages/viur/core/skeleton.py", line 370, in fromClient
    errors = _bone.fromClient(skelValues, key, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/achim/.local/share/virtualenvs/project-viur3-kGucZxTM/lib/python3.11/site-packages/viur/core/bones/base.py", line 530, in fromClient
    res, parseErrors = self.singleValueFromClient(parsedData, skel, name, data)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/achim/.local/share/virtualenvs/project-viur3-kGucZxTM/lib/python3.11/site-packages/viur/core/bones/date.py", line 206, in singleValueFromClient
    value = time_zone.localize(value)
            ^^^^^^^^^^^^^^^^^^
AttributeError: 'zoneinfo.ZoneInfo' object has no attribute 'localize'
```
Please re-verify. I think it's also better to discuss this change in a separate PR (my fault).

Comment (phorward) [reply core/bones/date.py:257]: Or is it a problem between Python 3.10 and 3.11?

Comment (phorward) [reply core/bones/date.py]: Hello, I've moved this part of the PR into a separate PR on #789. Will be closed here for now and was reverted.

Comment (sveneberth) [reply core/bones/date.py]: > Something is wrong. Did you test in on a clean environment?

I tested this in my existing python 3.10 env.

**Review by sveneberth — APPROVED** [review]

**Comment by phorward** [comment]
Ok thank you. I think we gonna merge this for now, but its not entirely tested.
Afterwards I would like to improve and generalize the test-suite so that several test cases on this bone can be created.

**Comment by sveneberth** [comment]
> Afterwards I would like to improve and generalize the test-suite so that several test cases on this bone can be created.

Take a look at https://github.com/viur-framework/viur-core/pull/797

**Pull request merged**